### PR TITLE
Report the final value of count_rewires back as provenance

### PR DESCRIPTION
**Pull request by andrewgait**
Fixes https://github.com/SpiNNakerManchester/sPyNNaker/issues/575 (I think) - is this a sensible value to be reporting back?

### CODE REVIEW

**Comment by andrewgait** [comment]
I think I've done what was suggested in the latest commit, but the numbers coming back for the structural plasticity scripts in PyNN8Examples seem lower than what I'd expect given your explanation... ?  Could probably do with someone checking them...

**Comment by pabogdan** [comment]
where can I find this provenance item?

**Comment by rowleya** [comment]
It should go in with the rest of the provenance items for a core, like timer tick overrunning etc.

**Comment by pabogdan** [comment]
Did something change with the default reporting? I'm not getting xmls out anymore in the docker I'm running this from.

**Comment by andrewgait** [comment]
The place I've been looking at the numbers is in the provenance database file (provenance.sqlite3 in the reports/../run_1/provenance_data directory).

**Comment by pabogdan** [comment]
Right. What is the recommended tool to look inside the database?

**Comment by rowleya** [comment]
The SQL is default; if you want to change back to XML, you would have to add ```provenance_format = xml``` to the ```[reports]``` section of the cfg file.

**Comment by andrewgait** [comment]
> Right. What is the recommended tool to look inside the database?

I've been using DB Browser for Sqlite (https://sqlitebrowser.org/) - other people may have other preferences, though.

**Comment by andrewgait** [comment]
> An integration test for this could be to start off with no connections and only enable rewiring with a formation rule. The network should run until all synapses are most likely created. This way you'd know if the reported number is correct.

That's a good idea - I'll get round to writing one at some point soon, in that case.

**Comment by andrewgait** [comment]
(I'll put it on an equivalent branch to this in sPyNNaker8 so that we can then merge them at the same time, and we get a further integration test check of this as a bonus).

**Comment by pabogdan** [comment]
> That's a good idea - I'll get round to writing one at some point soon, in that case.
Conversely, one could start with all-to-all connectivity, a removal only rule, and run until there are no connections left to remove. The number of rewires should be the same in both cases(forming up, removing down).

File: neural_modelling/src/neuron/c_main.c [hunk 1]
```diff
@@ -95,8 +96,8 @@ int32_t rewiring_period = 0;
 //! Flag representing whether rewiring is enabled
 bool rewiring = false;
 
-// FOR DEBUGGING!
-uint32_t count_rewires = 0;
+//! DEBGGING: for counting the number of rewiring attempts
```
Comment (alan-stokes) [inline neural_modelling/src/neuron/c_main.c:99]: if this is going into prov data. its no longer for debugging. its now for prov.

**Comment by alan-stokes** [comment]
with the comment changed. im fine with the changes

File: neural_modelling/src/neuron/c_main.c (hunk 1 above)
Comment (andrewgait) [reply neural_modelling/src/neuron/c_main.c]: Technically speaking this value is counting the number of attempts, whereas what goes into the provenance is the number of successful rewires.  But I have updated the comment anyway as there's no real need to say it's for debugging either.

**Pull request merged**